### PR TITLE
Feat/cloudinary

### PR DESCRIPTION
**Pull request by borislavkb**
- Add picture image upload functionality. 
- Add minor edits on the style on the images. Style on cards rendered via API to be edited. More CSS improvements coming soon with a different PR.
- Add image upload to match the render of the card.

### CODE REVIEW

File: src/components/ItemCard.css [hunk 1]
```diff
@@ -29,6 +33,12 @@
   border: 2px hidden;
   border-radius: 25px;
   padding: 0.25rem 0.2rem;
+  margin: 0.5rem;
+}
+
+.ItemCard__button:hover {
+  background-color: salmon;
+  color: black;
```
Comment (adrianammt) [inline src/components/ItemCard.css:41]: You could now define global variables for your colors. :)

File: src/pages/AddNewCake.css [hunk 2]
```diff
@@ -17,3 +17,9 @@
   letter-spacing: 0.1rem;
   color: whitesmoke;
```
Comment (adrianammt) [inline src/pages/AddNewCake.css:18]: Same with this color.

File: src/pages/AddNewCake.js [hunk 3]
```diff
@@ -12,6 +12,25 @@ export default function AddNewCake() {
...
+
+    const res = await fetch(
+      "https://api.cloudinary.com/v1_1/cakesapp/image/upload",
+      { method: "POST", body: data }
+    );
+    const file = await res.json();
+
+    setImage(file);
+  };
+
```
Comment (adrianammt) [inline src/pages/AddNewCake.js:33]: Is this what's uploading the images? Nice! What is that strange named string in line 22? The id?

File: src/pages/MyCakesList.js [hunk 4]
```diff
@@ -15,30 +15,36 @@ export default function MyCakesList() {
...
+  useEffect(() => {
+    localStorage.setItem("favoritesArray", JSON.stringify(listOfFavs));
+  }, [listOfFavs]);
+
   function handleDeleteItemFromList(id) {
     const updatedList = listOfCakes.filter((cake) => cake.id !== id);
     setListOfCakes(updatedList);
   }
-
+  // very slick solution from Adriana
```
Comment (adrianammt) [inline src/pages/MyCakesList.js:26]: 🤩 🦄 so cool! But you probably want to delete this comment before you merge.

File: src/components/Form.js [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
+      onChange={uploadImage}
+    >
       <i className="input-file--icon"></i>
 
       <input
         type="file"
         id="cakeImage"
         name="cakeImage"
         accept="image/png, image/jpeg"
+        onChange={uploadImage}
```
Comment (ScienceMaster7) [inline src/components/Form.js:15]: Is it necesary that you have the onChange twice ?
Wouldn't it be enough if you have either in the form or the input ?

**Review by ScienceMaster7 — COMMENTED** [review]
Well done 👍.
I left you a comment .

**Review by adrianammt — APPROVED** [review]
Awesome to see the deployment! It is looking good Boris! I have some little comments. Also I don't know what your plan for the Home page is, but if you are keeping those recipes there maybe you want to change the title to "Inspiration recipes" or something like that.

File: src/pages/AddNewCake.js (hunk 3 above)
Comment (borislavkb) [reply src/pages/AddNewCake.js:33]: This is taken from the upload settings from cloudinary. This is an interesting resource related to the topic 
https://medium.com/geekculture/how-to-upload-images-to-cloudinary-with-a-react-app-f0dcc357999c

and another one: 
https://cloudinary.com/documentation/image_upload_api_reference

File: src/pages/MyCakesList.js (hunk 4 above)
Comment (borislavkb) [reply src/pages/MyCakesList.js:26]: I will delete it. But this solution I "borrowed" from you and wanted to say thank you

File: src/components/Form.js (hunk 5 above)
Comment (borislavkb) [reply src/components/Form.js:15]: I see your point. You are right. In this current version I am not using the Form as a component. Will change it once I make the component working. :)

File: package.json [hunk 6]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^11.2.7",
     "@testing-library/user-event": "^12.8.3",
+    "axios": "^0.21.1",
```
Comment (namirsab) [inline package.json:9]: You intsalled axios but I see no usage of it. Let's remove it

Comment (borislavkb) [reply package.json:9]: Indeed, I was planning to use it but then found another way. Will be uninstalled with the next fix.

**Pull request merged**